Add OpenVPN infrastructure configuration

jattree opened the pull request:
Add Terraform configuration for deployed OpenVPN server:
- vpn-server.tf: EC2 instance (t3.small) and Elastic IP
- vpn-iam.tf: IAM roles and policies for AWS IAM authentication
- vpn-security-groups.tf: Security groups and VPN-based SSH access
- vpn-user-data.sh: Server initialization and OpenVPN setup script OpenVPN server running at 16.176.117.89 (ap-southeast Provides secure VPN access and enables SSH lockdown for other servers.

Commit:
Add OpenVPN infrastructure configuration

Add Terraform configuration for deployed OpenVPN server:
- vpn-server.tf: EC2 instance (t3.small) and Elastic IP
- vpn-iam.tf: IAM roles and policies for AWS IAM authentication
- vpn-security-groups.tf: Security groups and VPN-based SSH access
- vpn-user-data.sh: Server initialization and OpenVPN setup script
OpenVPN server running at 16.176.117.89 (ap-southeast-2)
Provides secure VPN access and enables SSH lockdown for other servers.

benrfairless commented on terraform/vpn-user-data.sh:
```diff
@@ -0,0 +1,183 @@
+set_var EASYRSA_REQ_PROVINCE   "NSW"
+set_var EASYRSA_REQ_CITY       "Sydney"
+set_var EASYRSA_REQ_ORG        "OAF"
+set_var EASYRSA_REQ_EMAIL      "admin@openaustraliafoundation.org.au"
```
Should this email be set to contact@oaf.org.au (or similar?)

```suggestion
set_var EASYRSA_REQ_EMAIL      "contact@oaf.org.au"
```

Commit:
Update OpenVPN configuration with working IAM authentication

- Add verify-client-cert none and script-security 2 to server config
- Update PAM config to use expose_authtok and seteuid flags
- Improve IAM auth script to use PAM_USER and better error logging
- Set AWS_DEFAULT_REGION for credential validation
- Enhance logging for easier troubleshooting

benrfairless commented on terraform/vpn-user-data.sh:
```diff
@@ -0,0 +1,193 @@
+
+# Validate credentials using STS GetCallerIdentity
+export AWS_ACCESS_KEY_ID="$ACCESS_KEY_ID"
+export AWS_SECRET_ACCESS_KEY="$SECRET_ACCESS_KEY"
```
Where are these keys kept/generated from?

benrfairless reviewed (commented):
Just a few other points:
- Why are we using Terraform for the provisioning of the server and not Ansible?
- Do we need to install Newrelic or any other monitoring on the server?
- Should the .tf files be in their own folder (`terraform/vpn` for example)?
- Are there any DNS changes required (such as creating a subdomain for VPN configuration).

Happy to discuss these changes on Slack given its a VPN/security change.

Br3nda commented on terraform/vpn-user-data.sh:
```diff
@@ -0,0 +1,193 @@
+cp pki/dh.pem /etc/openvpn/
+
+# Get server's private IP for routing
+PRIVATE_IP=$(curl -s http://169.254.169.254/latest/meta-data/local-ipv4)
```
this variable doesn't seem to be used.

Commit:
Refactor OpenVPN infrastructure to separate Terraform and Ansible

Separate concerns between infrastructure provisioning (Terraform) and
configuration management (Ansible) for the OpenVPN VPN server.

Terraform changes:
- Minimize vpn-user-data.sh to bootstrap only (Python installation)
- Update vpn-server.tf to use file() instead of templatefile()
- Infrastructure creates: EC2, IAM roles/policies, security groups, EIP

Ansible changes:
- Create internal/openvpn role for server configuration
- Implement IAM-based authentication via AWS STS
- Configure Easy-RSA PKI, OpenVPN service, and NAT routing
- Add openvpn group to inventory and site.yml playbook

Configuration:
- VPN network: 10.8.0.0/24
- VPC access: 172.31.0.0/16
- IAM group: vpn-users
- Region: ap-southeast-2

Br3nda reviewed (approved):
reviewed over zoom.
